Source/Urho3D/Core/StringUtils.cpp

```diff
@@ -744,8 +744,18 @@ unsigned GetStringListIndex(const char* value, const ea::string* strings, unsign
 
 unsigned GetStringListIndex(const char* value, const char* const* strings, unsigned defaultIndex, bool caseSensitive)
 {
-    unsigned i = 0;
+    // Allow value to be index instead of string representation.
+    if (*value >= '0' && *value <= '9')
+    {
+        char* end;
+        const unsigned res = std::strtoul(value, &end, 10);
+        if (res != ULONG_MAX)
+        {
+            return res;
```

Comment:
Hmm i very much do not like this. Now this function does two different things. What if i want to get an index of a string `"0abcd"` in the list? This function would break. IMO place for this code is in a separate function.

Reply:
In general I don't like how the code is enum serialization code is implemented as it doesn't check for bounds when taking string by index. Shall we use ea::array instead or maybe ea::span?

Reply:
Fixed